src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -170,14 +170,20 @@ def get(self, request: Request, project: Project) -> Response:
                 ):
                     derived = True
 
-                outcome = get_link(config, filepath, ctx["commit_id"])
-                # In some cases the stack root matches and it can either be that we have
-                # an invalid code mapping or that munging is expect it to work
-                if not outcome.get("sourceUrl"):
+                outcome = {}
+                munging_outcome = {}
+                # If the platform is a mobile language, get_link will never get the right URL without munging
+                if ctx["platform"] in ["java", "cocoa", "other"]:
                     munging_outcome = try_path_munging(config, filepath, ctx)
```

Comment:
Call `try_path_munging` instead of `get_link` the platform is a mobile language, as `get_link` will always fail for a mobile platform